directory/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,101 @@
+name: Directory
+files:
+- name: directory.yaml
+  options:
+  - template: init_config
+    options:
+    - name: directory
+      description: |
+        The directory to monitor.
+
+        On Windows, please make sure you escape back-slashes.
+        Otherwise, the YAML parser will fail (e.g "C:\\Users\\foo\\Downloads").
+      value:
+        type: string
+      required: true
+    - name: name
+      description: Tag metrics with a specified name.
+      value:
+        type: string
+        example: <DIRECTORY>
+    - name: dirtagname
+      description: |
+        The name of the key for the tag used for the directory, the value is the value of the `name` option.
+        The resulting tag attached is "<dirtagname>:<name>".
+      value:
+        type: string
+        example: <DIRECTORY_TAG_KEY>
+    - name: filetagname
+      description: |
+        The name of the key for the tag used for each file, the value is the filename.
+        The resulting tag is "<filetagname>:<filename>". The option is only used when `filegauges` is enabled
+        and if the directory contains 20 files or less (including sub-directories).
+      value:
+        type: string
+        example: <TAG_KEY_FILENAME>
+    - name: filegauges
+      description: When enabled, stats are an individual gauge per file (max. 20 files!) and not a histogram of the whole directory.
```

Comment:
Does this mean when `filegauges` is disabled, stats are a histogram of the whole directory?

Reply:
yes, default behavior iirc